test/drbgtest.c

```diff
@@ -824,66 +727,25 @@ static int test_rand_prediction_resistance(void)
         || !TEST_true(EVP_RAND_instantiate(z, 0, 0, NULL, 0, NULL)))
         goto err;
 
-    /*
-     * During a normal reseed, only the last DRBG in the chain should
-     * be reseeded.
-     */
-    inc_reseed_counter(y);
-    xreseed = reseed_counter(x);
-    yreseed = reseed_counter(y);
-    zreseed = reseed_counter(z);
-    if (!TEST_true(EVP_RAND_reseed(z, 0, NULL, 0, NULL, 0))
-        || !TEST_int_eq(reseed_counter(x), xreseed)
-        || !TEST_int_eq(reseed_counter(y), yreseed)
-        || !TEST_int_gt(reseed_counter(z), zreseed))
+    /* Fill the first buffer with random bytes */
+    if (!TEST_true(EVP_RAND_generate(z, buf1, sizeof(buf1), 0, 0, NULL, 0)))
         goto err;
 
     /*
      * When prediction resistance is requested, the request should be
      * propagated to the primary, so that the entire DRBG chain reseeds.
      */
-    zreseed = reseed_counter(z);
-    if (!TEST_true(EVP_RAND_reseed(z, 1, NULL, 0, NULL, 0))
-        || !TEST_int_gt(reseed_counter(x), xreseed)
-        || !TEST_int_gt(reseed_counter(y), yreseed)
-        || !TEST_int_gt(reseed_counter(z), zreseed))
-        goto err;
-
-    /*
-     * During a normal generate, only the last DRBG should be reseed */
-    inc_reseed_counter(y);
-    xreseed = reseed_counter(x);
-    yreseed = reseed_counter(y);
-    zreseed = reseed_counter(z);
-    if (!TEST_true(EVP_RAND_generate(z, buf1, sizeof(buf1), 0, 0, NULL, 0))
-        || !TEST_int_eq(reseed_counter(x), xreseed)
-        || !TEST_int_eq(reseed_counter(y), yreseed)
-        || !TEST_int_gt(reseed_counter(z), zreseed))
+    if (!TEST_true(EVP_RAND_reseed(z, 1, NULL, 0, NULL, 0)))
         goto err;
 
     /*
      * When a prediction resistant generate is requested, the request
      * should be propagated to the primary, reseeding the entire DRBG chain.
      */
```

Comment:
It is not obvious, even after several readings, what the difference is between this comment and the comment in lines 734-737